Foundry Tests "Panic" Crashing

Several tests this evening have crashed. Could somebody do some research on resolving this?

```
Run yarn test:all
➤ YN0000: No files changed, compilation skipped
➤ YN0000: The application panicked (crashed).
➤ YN0000: Message:  Unable to create fork: 
➤ YN0000: Could not instantiate forked environment with fork url: https://rpc.ankr.com/eth
➤ YN0000: Location: evm/src/executor/backend/mod.rs:[4](https://github.com/ubiquity/ubiquity-dollar/actions/runs/3622649352/jobs/6107615296#step:10:5)09
➤ YN0000: 
➤ YN0000: This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry
➤ YN0000: 
➤ YN0000: Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
➤ YN0000: Run with RUST_BACKTRACE=full to include source snippets.
➤ YN0000: Done in 3s 63[5](https://github.com/ubiquity/ubiquity-dollar/actions/runs/3622649352/jobs/6107615296#step:10:6)ms
Error: Process completed with exit code 1.
```

<img width="739" alt="image" src="https://user-images.githubusercontent.com/4975670/206020025-553e6d6c-fb03-47fc-bff7-df4b1cd705a2.png">

https://github.com/ubiquity/ubiquity-dollar/actions/runs/3622649352/jobs/6107615296

--- a/packages/contracts/package.json
+++ b/packages/contracts/package.json
@@ -23,7 +23,7 @@
   "scripts": {
     "forge": "forge",
     "build": "forge build",
-    "test": "forge test --fork-url https://cloudflare-eth.com",
+    "test": "ts-node ./scripts/runner/runner.ts",
     "slither:run": "slither . --compile-force-framework foundry",
     "echidna": "echidna-test . --config echidna.config.yml",
     "prettier": "forge fmt",

--- a/packages/contracts/scripts/runner/conf.ts
+++ b/packages/contracts/scripts/runner/conf.ts
@@ -1,0 +1,20 @@
+export const RPC_DELAY = 300;
+
+export const RPC_LIST: string[] = [
+    "https://ethereum.publicnode.com",
+    "https://cloudflare-eth.com",
+    "https://rpc.ankr.com/eth",
+    "https://eth-mainnet.public.blastapi.io",
+    "https://eth-mainnet-public.unifra.io",
+    "https://nodes.mewapi.io/rpc/eth",
+    "https://rpc.flashbots.net"
+];
+
+export const REQ_BODY = {
+    method: "eth_blockNumber",
+    params: [],
+    id: "1",
+    jsonrpc: "2.0"
+};
+
+export const RESP_STATUS = 200;

--- a/packages/contracts/scripts/runner/runner.ts
+++ b/packages/contracts/scripts/runner/runner.ts
@@ -1,0 +1,40 @@
+import axios from "axios";
+import { spawn } from "child_process";
+import {
+    RPC_LIST,
+    RPC_DELAY,
+    REQ_BODY,
+    RESP_STATUS
+} from "./conf";
+
+const getRandom = async (array: string[]) => {
+    const length = array == null ? 0 : array.length;
+    return length ? array[Math.floor(Math.random() * length)] : undefined;
+};
+
+const getUrl = async () => {
+    let isRPC = false;
+
+    while (!isRPC) {
+        try {
+            const rpcUrl = await getRandom(RPC_LIST);
+            const resp = await axios.post(rpcUrl as string, REQ_BODY);
+
+            if (resp.status === RESP_STATUS) {
+                isRPC = true;
+                return rpcUrl;
+            }
+        } catch (error) {
+            await setTimeout(() => ({}), RPC_DELAY);
+        }
+    }
+};
+
+(async () => {
+    const currentRPC = await getUrl();
+    console.log(`using ${currentRPC} for testing...`);
+    const command = spawn("forge", ["test", "--fork-url", currentRPC as string]);
+    command.stdout.on("data", (output: any) => {
+        console.log(output.toString());
+    });
+})();

